core/lib/zksync_core/src/consistency_checker/mod.rs

```diff
@@ -504,11 +504,21 @@ impl ConsistencyChecker {
             .0
             .saturating_sub(self.max_batches_to_recheck)
             .into();
+
+        let last_processed_batch = self
+            .pool
+            .connection()
+            .await?
+            .blocks_dal()
+            .get_consistency_checker_last_processed_l1_batch()
+            .await?;
+
         // We shouldn't check batches not present in the storage, and skip the genesis batch since
         // it's not committed on L1.
         let first_batch_to_check = first_batch_to_check
             .max(earliest_l1_batch_number)
-            .max(L1BatchNumber(1));
+            .max(L1BatchNumber(1))
+            .max(last_processed_batch);
```

Comment:
It should be the next batch after the processed one, right? Would also allow skipping `max(L1BatchNumber(1))` altogether.

Reply:
Yes; corrected